File: android/src/main/cn/io/github/nullptrx/pangleflutter/util/MessageUtils.java
package io.github.nullptrx.pangleflutter.util;

import io.flutter.plugin.common.MethodChannel;
import io.github.nullptrx.pangleflutter.bean.PangleResult;

public class MessageUtils {

  public static void postSimpleMessage(MethodChannel.Result result, int code, String message) {
    PangleResult pangleResult = new PangleResult();
    pangleResult.code = code;
    pangleResult.message = message;
    postCustomMessage(result, pangleResult);
  }

  public static void postVerifyMessage(MethodChannel.Result result, int code, String message,
      Boolean isVerify) {
    PangleResult pangleResult = new PangleResult();
    pangleResult.code = code;
    pangleResult.message = message;
    pangleResult.verify = isVerify;
    postCustomMessage(result, pangleResult);
  }

  public static void postCustomMessage(MethodChannel.Result result, PangleResult pangleResult) {
    if (result != null && pangleResult != null) {
      try {
        result.success(pangleResult.toMap());
      } catch (Exception ignored) {
      }
    }
  }
}

File: android/src/main/cn/io/github/nullptrx/pangleflutter/util/MessageUtils.kt
package io.github.nullptrx.pangleflutter.util

import io.flutter.plugin.common.MethodChannel.Result
import io.github.nullptrx.pangleflutter.bean.PangleResult

object MessageUtils {
  @JvmStatic fun postSimpleMessage(
    result: Result?,
    code: Int,
    message: String?
  ) {
    val pangleResult = PangleResult()
    pangleResult.code = code
    pangleResult.message = message
    postCustomMessage(result, pangleResult)
  }

  @JvmStatic fun postVerifyMessage(
    result: Result?,
    code: Int,
    message: String?,
    isVerify: Boolean?
  ) {
    val pangleResult = PangleResult()
    pangleResult.code = code
    pangleResult.message = message
    pangleResult.verify = isVerify
    postCustomMessage(result, pangleResult)
  }

  @JvmStatic fun postCustomMessage(
    result: Result?,
    pangleResult: PangleResult?
  ) {
    if (result != null && pangleResult != null) {
      try {
        result.success(pangleResult.toMap())
      } catch (ignored: Exception) {
      }
    }
  }
}
